contributor-docs/javascript.md

```diff
@@ -0,0 +1,33 @@
+# Working with JavaScript
+
+Some Primer ViewComponents rely on JavaScript for client-side behaviors. [Read more about the introduction of behavioral components in this ADR](https://github.com/primer/view_components/blob/main/adr/0250-developing-and-publishing-clientside-behaviours.md).
+
+## JavaScript strategy
+
+Alongside the `primer_view_components` gem we release a `@primer/view-components` npm package which bundles all the JavaScript needed for components with client-side behaviors.
+
+We write JavaScript in the Primer ViewComponents repository alongside its Ruby functionality, but some components import their behaviors from other npm packages:
+
+### GitHub Elements
+
+GitHub’s [open-source web components](https://github.com/github/github-elements) are accessible unstyled elements that are released as separate npm packages and are owned by the Web Systems team.
+
+Initially all our client-side behaviors came from using these web components but we found a lot of friction when trying to update their functionality and we decided we don’t want to create new web components in GitHub’s collection solely for Primer use. Only use GitHub Elements when there’s already an element that matches your use case without needing changes.
```

Comment:
I am not sure about this part:

>  we decided we don’t want to create new web components in GitHub’s collection solely for Primer use.

I think we didn't have clear answers on where _new_ custom elements created for Primer use should live. We recently added [tool-tip.ts ](https://github.com/primer/view_components/blob/main/app/components/primer/alpha/tooltip.ts)to PVC with the idea we can move it elsewhere if we find a better home for it, and I think we might do the same with dialog currently living in dotcom. The accessibility team is also working on a new action menu custom element as well for Primer use so I am under impression we can keep making custom elements.

Reply:
@keithamus I meant friction around co-developing a ViewComponent alongside web component changes (having to yarn link them and deal with releases etc.) - nothing that you could improve, sorry if that came off snarky here! I think there's also some murk around creating new web components for Primer because of ownership, are they then your team's responsibility or Design Infra's or Accessibility's..

I was off sick but it was talked about in UI Club on Feb 10th and I thought this reflected what you concluded. @joelhawksley can you weigh in? I think this is the direction that we've talked about going in with JS in PVC, not necessarily because it's a clear winner but that we'd rather have _some_ guidance for these situations, and, as Kate says, we can easily move the JS from PVC later if we want to go in a different direction.

Reply:
@pouretrebelle thanks for the ping! I don't recall any specific conclusion from UI Club on 2/10. I do agree that we need to focus on contribution workflows, both for CSS and JS.

I think we're right to start by authoring JS alongside PVCs. If it's needed elsewhere, we can address that process at that point.

Perhaps we could soften the language in these docs to say something like "we should try to leverage existing GitHub elements when possible. If they don't fit the bill, write custom JS in PVC..."

Reply:
Just my $0.02: we're also planning to move `NavigationList` from dotcom into PVC pretty soon, which comes with its own Javascript behaviors that I don't think are covered by GitHub Elements.